sources/packages/backend/apps/api/src/auth/guards/institution/institution-student-data-access.guard.ts

```diff
@@ -37,7 +42,13 @@ export class InstitutionStudentDataAccessGuard implements CanActivate {
           user.authorizations.institutionId,
           studentId,
         );
-      return hasStudentDataAccess;
+      if (hasStudentDataAccess) {
+        return true;
+      }
+      // The institution does not have access to student data.
+      throw new ForbiddenException(
+        "The institution is not allowed access the student data of given student.",
```

Comment:
Minor suggestion: "The institution is not allowed access to the student data of the given student."